File: dotnet/samples/02-agents/providers/OpenAIProvider.cs
// Copyright (c) Microsoft. All rights reserved.

// Provider: OpenAI
// Create an agent using OpenAI Chat Completion as the backend.
//
// For docs: https://learn.microsoft.com/agent-framework/agents/providers

using Microsoft.Agents.AI;
using Microsoft.Extensions.AI;
using OpenAI;
using OpenAI.Chat;

var apiKey = Environment.GetEnvironmentVariable("OPENAI_API_KEY")
    ?? throw new InvalidOperationException("OPENAI_API_KEY is not set.");
var model = Environment.GetEnvironmentVariable("OPENAI_MODEL") ?? "gpt-4o-mini";

// <openai>
AIAgent agent = new OpenAIClient(apiKey)
    .GetChatClient(model)
    .AsAIAgent(instructions: "You are good at telling jokes.", name: "Joker");
// </openai>

Console.WriteLine(await agent.RunAsync("Tell me a joke about a pirate."));

File: python/samples/02-agents/providers/openai_provider.py
# Copyright (c) Microsoft. All rights reserved.

import asyncio

from agent_framework.openai import OpenAIResponsesClient

"""
OpenAI Provider

Demonstrates setting up OpenAIResponsesClient and running a simple query.

Environment variables:
- OPENAI_API_KEY: Your OpenAI API key
- OPENAI_RESPONSES_MODEL_ID: Model to use (default: gpt-4.1-mini)

For more OpenAI examples:
- With tools: getting_started/agents/openai/openai_responses_client_with_function_tools.py
- With structured output: ../structured_output.py
- Docs: https://learn.microsoft.com/agent-framework/providers/openai
"""


async def main() -> None:
    print("=== OpenAI Provider ===\n")

    # <create_agent>
    agent = OpenAIResponsesClient().as_agent(
        name="OpenAIAgent",
        instructions="You are a helpful assistant.",
    )
    # </create_agent>

    # <run_query>
    query = "What is the capital of France?"
    print(f"User: {query}")
    response = await agent.run(query)
    print(f"Agent: {response}")
    # </run_query>


if __name__ == "__main__":
    asyncio.run(main())
